liveview-android/src/main/java/org/phoenixframework/liveview/data/dto/PagerDTO.kt

```diff
@@ -0,0 +1,368 @@
+        composableNode: ComposableTreeNode?, paddingValues: PaddingValues?, pushEvent: PushEvent
+    ) {
+        val childrenNodes = composableNode?.children ?: emptyArray()
+        val state = rememberPagerState(
+            initialPage = currentPage,
+            initialPageOffsetFraction = initialPageOffsetFraction ?: 0f,
+            pageCount = { pageCount },
+        )
+        when (composableNode?.node?.tag) {
+            ComposableTypes.horizontalPager -> {
+                HorizontalPager(
+                    state = state,
+                    modifier = modifier,
+                    contentPadding = contentPadding ?: PaddingValues(0.dp),
+                    pageSize = pageSize ?: PageSize.Fill,
+                    beyondBoundsPageCount = beyondBoundsPageCount
+                        ?: PagerDefaults.BeyondBoundsPageCount,
+                    pageSpacing = pageSpacing,
+                    verticalAlignment = verticalAlignment ?: Alignment.CenterVertically,
+                    // TODO flingBehavior: SnapFlingBehavior,
+                    userScrollEnabled = userScrollEnabled,
+                    reverseLayout = reverseLayout,
+                    key = { index ->
+                        childrenNodes[index].id
+                    },
+                    // TODO pageNestedScrollConnection: NestedScrollConnection,
+                    pageContent = { index ->
+                        PhxLiveView(childrenNodes[index], pushEvent, composableNode, null, this)
+                    }
+                )
+            }
+
+            ComposableTypes.verticalPager -> {
+                VerticalPager(
+                    state = state,
+                    modifier = modifier,
+                    contentPadding = contentPadding ?: PaddingValues(0.dp),
+                    pageSize = pageSize ?: PageSize.Fill,
+                    beyondBoundsPageCount = beyondBoundsPageCount
+                        ?: PagerDefaults.BeyondBoundsPageCount,
+                    pageSpacing = pageSpacing,
+                    horizontalAlignment = horizontalAlignment ?: Alignment.CenterHorizontally,
+                    // TODO flingBehavior: SnapFlingBehavior,
+                    userScrollEnabled = userScrollEnabled,
+                    reverseLayout = reverseLayout,
+                    key = { index ->
+                        childrenNodes[index].id
+                    },
+                    // TODO pageNestedScrollConnection: NestedScrollConnection,
+                    pageContent = { index ->
+                        PhxLiveView(childrenNodes[index], pushEvent, composableNode, null, this)
+                    }
+                )
+            }
+        }
+        LaunchedEffect(state) {
+            snapshotFlow {
+                state.currentPage
+            }
+                .distinctUntilChanged()
+                // debouncing to avoid send intermediate values between the scrolling animation
+                .debounce(300)
```

Comment:
Could this use the `phx-debounce` attribute for its value if present?

Reply:
I don't think so... This code avoids the scenario where the pager is used together with the `Tab` component (as shown in the GIF above). Let's say the first tab is selected, and then, the user selects the third tab, an animation happens sliding from the first to the third tab (which shows the second tab quickly during the animation). So this code avoids sending an update informing that the current page changed from 0 to 1, and then another update sending a change from 1 to 2. Because it's actually changing from 0 to 2.